docs/feature-gates.md

```diff
@@ -341,3 +342,14 @@ More documentation will be coming in the future.
 
 User should provide a file server with this feature, and store its authentication credential in a Secret. Antrea
 Controller is required to be configured with the permission to read the Secret.
+
+### L7NetworkPolicy
+
+`L7NetworkPolicy` enables users to protect their applications by specifying how they are allowed to communicate with
+others with application context, providing fine-grained control over the network traffic beyond IP, transport protocol,
+and port. Refer to this [document](./antrea-l7-network-policy.md) for more information.
```

Comment:
nit: we don't usually add the `/.` to the link when the target is in the same directory